src/core_plugins/kibana/public/visualize/editor/editor.js

```diff
@@ -114,19 +129,22 @@ uiModules
       vis: savedVisState
     };
 
-    $state = new AppState(stateDefaults);
+    // This is used to generate a 'uiState' PersistedState instance, and
+    // implictly add 'change' and 'fetch_with_changes' event handlers.
+    const appState = new AppState(stateDefaults);
 
-    if (!angular.equals($state.vis, savedVisState)) {
+    // NOTE: Why would appState.vis not equal the savedVisState?
```

Comment:
The `savedVis` is pulled from elasticsearch, but the `appState` is pulled from the url, with the defaults applied. If the url was from a previous session which included modifications to the appState then they won't be equal.